### PR TITLE
CLSContext now supports continuations with async-await

### PR DESCRIPTION
**Pull request by Ghost---Shadow**
`continuation-local-storage` has issues with `async-await` so someone made a library called [cls-hooked](https://www.npmjs.com/package/cls-hooked) which fixes that problem.

This PR merely changes the dependency from `cls` to `cls-hooked`.

Also, added a test case to demonstrate the phenomenon. 

Possible duplicate #135

### CODE REVIEW

File: packages/zipkin-context-cls/test/CLSContext.test.js [hunk 1]
```diff
@@ -60,4 +60,25 @@ describe('CLSContext', () => {
...
+
+    async function afn() {
+      const cid1 = await log(1);
+      const cid2 = await log(2);
+      return {cid1, cid2};
+    }
+
+    async function newP(id) {
+      const {cid1, cid2} = await ctx.letContext(id, () => afn());
+      // console.log(cid1, cid2);
```
Comment (adriancole) [inline packages/zipkin-context-cls/test/CLSContext.test.js]: remove the commented out stuff

File: packages/zipkin-context-cls/test/.eslintrc [hunk 2]
```diff
@@ -4,5 +4,6 @@
   },
   "globals": {
     "expect": true
-  }
-}
+  },
+  "parser": "babel-eslint"
```
Comment (adriancole) [inline packages/zipkin-context-cls/test/.eslintrc]: is this change necessary?

File: package.json [hunk 3]
```diff
@@ -33,4 +33,4 @@
   "workspaces": [
     "packages/*"
   ]
-}
+}
```
Comment (adriancole) [inline package.json]: might want to change your settings to ensure EOL before EOF

File: packages/zipkin-context-cls/test/CLSContext.test.js (hunk 1 above)
Comment (Ghost---Shadow) [reply packages/zipkin-context-cls/test/CLSContext.test.js]: Ok.

File: package.json (hunk 3 above)
Comment (Ghost---Shadow) [reply package.json]: .json files dont have EOL before EOF. Only js files do. I will add them if you want to.

File: packages/zipkin-context-cls/test/.eslintrc (hunk 2 above)
Comment (Ghost---Shadow) [reply packages/zipkin-context-cls/test/.eslintrc]: Yeah, eslint was freaking out.

**Comment by Ghost---Shadow** [comment]
Going through #135 I was thinking what about creating a new folder `zipkin-context-cls-hooked` and leaving the original `zipkin-context-cls` as it is. 

What do you think?

**Comment by adriancole** [comment]
@Ghost---Shadow just to keep us from hopping issues too much. what's your idea on doing that? ex the why not the what. I know we've had some debates about how to keep the build running in various fashions and a lot of people have participated in these discussions. Would be nice to summarize work before this, what it solves from previous attempts, what it doesn't solve, how it addresses the build concerns, any compat impact with regards to react native or browser (like angular) apps.

**Comment by Ghost---Shadow** [comment]
@adriancole I am afraid that I am not knowledgeable enough to have opinions on this subject matter. 

However, I urgently need this feature for a project so I created a new npm package called [zipkin-context-cls-hooked](https://www.npmjs.com/package/zipkin-context-cls-hooked).

1. If you decide it to merge this as `zipkin-context-cls`. Then I will deprecate my package and link to this repo.
2. If you decide to create a new package `zipkin-context-cls-hooked` then I will unpublish my package and publish the one in this repo, so that npm points to the right thing.

**Comment by evan-scott-zocdoc** [comment]
Would this fix https://github.com/openzipkin/zipkin-js/issues/88#issuecomment-289325561 in terms of how express async middlewares are handled via CLSContext?

**Comment by Ghost---Shadow** [comment]
Give it a try.

**Comment by justlaputa** [comment]
Hi I'm coming to this issue because I met same problem, may I ask how is this issue now? it will get merged?

**Comment by adriancole** [comment]
@eirslett @DanielMSchmidt considering this: https://medium.com/airbnb-engineering/sunsetting-react-native-1868ba28e30a

do we continue to support (watch out for conflicts with) react native? This topic has locked up a number of issues I'm afraid.

**Comment by DanielMSchmidt** [comment]
@adriancole I would not say React Native is dying just because AirBnB decides to move away from it. That being said, we only have one version of zipkin.js that actually works in a none node environment, the current versions are broken anyway, so why care 🤷‍♂️ 

I haven't had the time and energy to put up meaningful E2E tests for either the browser or React Native (I think having one of the two would be a good start), sorry about that.

**Comment by jcchavezs** [comment]
Any chance to continue with this @Ghost---Shadow?

**Comment by Ghost---Shadow** [comment]
@jcchavezs I dont even remember what this was :P 

I guess not. I have moved on.

**Comment by r4j4h** [comment]
@Ghost---Shadow  where does your fork's code live? The npm page has this project set as its home page and both Google search and Github search are failing me.

zipkin-context-cls-hooked's npm readme even instructs using "zipkin-context-cls", so it feels rather hastily set up and unmaintained. There was conversation https://github.com/openzipkin/zipkin-js/pull/135#issuecomment-335475695 to have it live in https://github.com/openzipkin-contrib but this has not been done.

Just curious about the actual state of this. :)

**Comment by Ghost---Shadow** [comment]
@r4j4h I just forked and fixed it for my personal use, I dont care what others are doing :)

https://github.com/Ghost---Shadow/zipkin-js/tree/bugfix/asyc-await

**Comment by Ghost---Shadow** [comment]
As promised I have deprecated the package

https://www.npmjs.com/package/zipkin-context-cls-hooked

**Comment by jcchavezs** [comment]
Might be too much to ask but would you check if someone is using it and
open PRs to that? Do you use it yourself?

**Comment by Ghost---Shadow** [comment]
@jcchavezs I dont use it anymore. There are 200+ weekly downloads. People are using that.

They should be notified the next time they do an `npm install`
